presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -747,9 +753,24 @@ public synchronized Map<String, Optional<Partition>> getPartitionsByNames(Metast
         }
         Map<String, Optional<Partition>> delegateResult = delegate.getPartitionsByNames(metastoreContext, databaseName, tableName, partitionNamesToQuery.build());
         resultBuilder.putAll(delegateResult);
+
+        cacheLastDataCommitTimes(delegateResult, databaseName, tableName);
+
         return resultBuilder.build();
     }
 
+    public void cacheLastDataCommitTimes(Map<String, Optional<Partition>> existingPartitions, String databaseName, String tableName)
+    {
+        ImmutableList<DateTime> lastDataCommitTimes = existingPartitions.values().stream()
+                .filter(Optional::isPresent)
+                .map(partition -> partition.get().getLastDataCommitTime())
+                .map(commitTime -> new DateTime(TimeUnit.SECONDS.toMillis(commitTime)))
```

Comment:
To save memory, probably we don't mind using long instead of `DateTime`. If that is the case, we can use `long[]` instead of `List<DateTime>`.

Reply:
same: https://github.com/prestodb/presto/pull/17830/files#r927419444